app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -503,8 +513,18 @@ def vm_instance_defaults
         :volumes    => [new_volume],
         :scsi_controller => { :type => scsi_controller_default_type },
         :datacenter => datacenter,
+        :firmware => 'automatic',
         :boot_order => ['network', 'disk']
       )
     end
+
+    def firmware_mapping(firmware)
+      case firmware
+      when /UEFI/
```

Comment:
Scratch the SecureBoot comment, there is "UEFI" there so it's valid. But I'd suggest to move it and add a unit test.